source/whats-new.txt

```diff
@@ -22,6 +22,305 @@ What's New in 4.0
 
 New features of the 4.0 Node.js driver release include:
 
+TypeScript
+~~~~~~~~~~
+
+- We've migrated the driver to TypeScript. You can now harness the type
+  hinting and intellisense features in editors that support it to develop
+  your MongoDB applications. Enjoy the benefits of this work in pure JavaScript
+  projects as well!
+
+- The underlying BSON library used by this version is now migrated to
+  TypeScript.
+
+- Inline documentation is now consistenly formatted to improve display
+  in editors.
+
+- If you are a user of the community types ``@types/mongodb``, there will
+  **likely be issues** adopting the types from our codebase. We could not
+  achieve a one to one match in types due to the details of writing the
+  codebase in TypeScript.
+
+  We'd love to hear your TypeScript related feature requests! Please submit
+  ideas on our `JIRA project here <https://jira.mongodb.org/projects/NODE>`__.
+
+Key Changes
+~~~~~~~~~~~
+
+Node.js Version
++++++++++++++++
+
+The minimum supported version of Node.js is now v12.9 or greater for
+version 4 of the driver. Support for our 3.x branches will continue
+until summer 2022 to allow time to upgrade.
+
+.. note::
+
+   3.x supports back to Node.js v4.
+
+
+Cursor Improvements
++++++++++++++++++++
+
+Affected classes:
```

Comment:
Issue: I don't know/understand how these classes were affected. I think there needs to be some wording to clarify.